File: araok/src/main/java/ru/otus/dto/ContentMediaIdDto.java
package ru.otus.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.otus.domain.ContentMediaId;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ContentMediaIdDto {
    private ContentDto content;

    private MediaTypeDto mediaType;

    public static ContentMediaId toDomainObject(ContentMediaIdDto dto) {
        return ContentMediaId.builder()
                .content(
                        ContentDto.toDomainObject(dto.content)
                )
                .mediaType(
                        MediaTypeDto.toDomainObject(dto.mediaType)
                )
                .build();
    }

    public static ContentMediaIdDto toDto(ContentMediaId contentMediaId) {
        return ContentMediaIdDto.builder()
                .content(
                        ContentDto.toDto(contentMediaId.getContent())
                )
                .mediaType(
                        MediaTypeDto.toDto(contentMediaId.getMediaType())
                )
                .build();
    }
}

File: araok-android/src/main/java/ru/otus/data/dto/ContentMediaIdDto.kt
package ru.araok.data.dto

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import ru.araok.entites.ContentMediaId

@JsonClass(generateAdapter = true)
data class ContentMediaIdDto(
    @Json(name = "content") override val content: ContentDto,
    @Json(name = "mediaType") override val mediaType: MediaTypeDto
): ContentMediaId
